internal/client/client.go

```diff
@@ -1,14 +1,27 @@
 package client
 
 import (
+	"crypto/tls"
+	"net/http"
+
 	"github.com/axiomhq/axiom-go/axiom"
 
 	"github.com/axiomhq/cli/pkg/version"
 )
 
 // New returns a new Axiom client.
-func New(baseURL, accessToken, orgID string, options ...axiom.Option) (*axiom.Client, error) {
+func New(baseURL, accessToken, orgID string, insecure bool, options ...axiom.Option) (*axiom.Client, error) {
+	httpClient := axiom.DefaultHTTPClient()
+
+	if insecure {
+		if transport, ok := httpClient.Transport.(*http.Transport); ok {
+			// nolint:gosec
+			transport.TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+		}
```

Comment:
Could use a warning or even error if it couldn't set the TLS config

Reply:
Kinda agree, but this would require some more changes I'm not sure about. I mean, as long as we don't do crazy stuff with the default http.Client, this will work.

Reply:
Thoughts?

Reply:
I think since we control the http client we can drop the `, ok` which will panic if it's not a `*http.Transport`

Reply:
I agree, but I would like to have proper tests first. When we change the client in `axiom-go`, we could introduce a panic without actually capturing it with a test. Can we agree on merging but opening an issue about this?

Reply:
Sounds like a plan 👍